contracts/protocol/libraries/logic/AuctionLiquidationLogic.sol

```diff
@@ -0,0 +1,181 @@
+        mapping(address => DataTypes.UserConfigurationMap) storage usersConfig,
+        DataTypes.ExecuteAuctionLiquidationEndParams memory params
+    ) external {
+        AuctionLiquidationEndLocalVars memory vars;
+
+        DataTypes.Auction storage _auction = auctionData[
+            params.collateralAsset
+        ][params.collateralTokenId];
+
+        // check if the auction is still going - check if time is up
+        require(
+            block.timestamp > _auction.endTime,
+            "Auction end time not reached"
+        );
+
+        // check if is still pending
+        require(!_auction.isResolved, "Auction is already resolved");
+
+        // set resolved
+        _auction.isResolved = true;
+
+        // check caller is highest bid
+        require(
+            params.bidder == _auction.highestBidder,
+            "Caller is not the highest bidder"
+        );
+
+        // check highest bid is above 95% of floor
+        vars.collateralPrice = IPriceOracleGetter(params.priceOracle)
+            .getAssetPrice(params.collateralAsset);
+        require(
+            _auction.highestBid > (95 * vars.collateralPrice) / 100,
+            "Highest bid is not above 95% of floor"
+        );
+
+        // check that collateral owner is still underwater
+        DataTypes.UserConfigurationMap storage userConfig = usersConfig[
+            params.user
+        ];
+
+        (, , , , , , , vars.healthFactor, , ) = GenericLogic
+            .calculateUserAccountData(
+                reservesData,
+                reservesList,
+                DataTypes.CalculateUserAccountDataParams({
+                    userConfig: userConfig,
+                    reservesCount: params.reservesCount,
+                    user: params.user,
+                    oracle: params.priceOracle
+                })
+            );
+
+        require(
+            vars.healthFactor < THRESHOLD,
+            "Users health factor must be below the threshold"
+        );
+
+        /// TODO: settle debt from liquidation
+
+        // repay debt
+        // or supply -> wrapped eth gw
+        // also pay fee
+    }
+
+    /// @notice people should be able to bid
+    function executeAuctionLiquidationBid(
+        mapping(address => mapping(uint256 => DataTypes.Auction))
+            storage auctionData,
+        DataTypes.ExecuteAuctionLiquidationBidParams memory params
+    ) external {
+        DataTypes.Auction storage _auction = auctionData[
+            params.collateralAsset
+        ][params.collateralTokenId];
+
+        require(
+            params.bidAmount > MIN_BID_THRESHOLD,
+            "Bid is too small, please increase"
+        );
+
+        require(
+            params.bidAmount > _auction.highestBid,
+            "Bid is not higher than existing highest bid"
+        );
+
+        // return previous amount to last highest bidder
+        payable(_auction.highestBidder).transfer(_auction.highestBid);
```

Comment:
@WalidOfNow can you help me determine the which contract the ETH should be moved in and out from? Maybe this should be on the pool?